pysumma/simulation.py

```diff
@@ -41,190 +40,74 @@ def __init__(self, executable, filemanager, run_suffix='_pysumma_run'):
         self.local_attributes = self.manager.local_attributes
         self._status = 'Initialized'
 
-    def exec_local(self, run_suffix):
-        if self.summa_code is not None:
-            cmd = "{} -p never -s {} -m {}".format(
-                self.executable, run_suffix, self.manager_path)
-            # run shell script in python and print output
-            cmd = shlex.split(cmd)
-            p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-            output = p.communicate()[0].decode('utf-8')
-            print(output)
-            if 'FATAL ERROR' in output:
-                raise Exception("SUMMA failed to execute!")
-            # define output file name as sopron version of summa
-            out_file_path = (self.manager.output_path.value
-                             + self.manager.output_prefix.value + '_output_'
-                             + run_suffix + '_timestep.nc')
+    def gen_summa_cmd(self, processes=1, prerun_cmds=[],
+                  startGRU=None, countGRU=None, iHRU=None, freq_restart=None,
+                  progress='m'):
+        prerun_cmds.append('export OMP_NUM_THREADS={}'.format(processes))
```

Comment:
Even though I didn't use `prerun_cmds`, it showed up in cmd command for SUMMA. So if there is `export OMP_NUM_THREADS=' inside SUMMA cmd, an error occurred.
Change to 
if prerun_cmds != []:  
            prerun_cmds.append('export OMP_NUM_THREADS={}'.format(processes))

Reply:
This makes it explicit if SUMMA is being run with multithreading, and I think is correct. I am updating the docker run to fix this.

Reply:
<Command>
docker run -v /media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/settings/wrrPaperTestCases/figure07:/media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/settings/wrrPaperTestCases/figure07 -v /media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/settings/:/media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/settings/ -v /media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/testCases_data/inputData/fieldData/reynolds/:/media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/testCases_data/inputData/fieldData/reynolds/ -v /media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/output/wrrPaperTestCases/figure07/:/media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/output/wrrPaperTestCases/figure07/export OMP_NUM_THREADS=1 && export OMP_NUM_THREADS=1 &&  uwhydro/summa:sopron_2018 -p never -s hs -m /media/sf_pysumma/a0105d479c334764ba84633c5b9c1c01/a0105d479c334764ba84633c5b9c1c01/data/contents/summaTestCases_2.x/settings/wrrPaperTestCases/figure07/summa_fileManager_riparianAspenSimpleResistance.txt
<Error messag>
docker: invalid reference format: repository name must be lowercase.
See 'docker run --help'.

There's still an error here. As I told before, we need if statement for this.
if prerun_cmds != []:
      prerun_cmds.append('export OMP_NUM_THREADS={}'.format(processes))